.github/workflows/test.yml

```diff
@@ -21,4 +21,4 @@ jobs:
     uses: ./.github/workflows/test_template.yml
     with:
       runs-on: '["ubuntu-latest", "macos-latest", "windows-latest"]'
-      python-version: '["3.8", "3.9", "3.10", 3.11]'
+      python-version: '["3.9", "3.10", 3.11]'
```

Comment:
@skoudoro So we've lost 3.8 here. Re: https://github.com/dipy/dipy/pull/2715#discussion_r1348580722. Will it be brought back?

Reply:
> Will it be brought back?

No, I do not think so.

I might add it in here, https://github.com/dipy/dipy/blob/268b7182c273808e9d6eadb8e90c591a42ef6e02/.github/workflows/test_compat.yml#L14. but just until december.

Reply:
> No, I do not think so.

Is there a particular reason? Is `trx` not compatible with 3.8 or is it that it was not tested? According to https://github.com/tee-ar-ex/trx-python/blob/master/.github/workflows/python-package.yml#L15, it is. Were DIPY tests failing for Python 3.8? Was it hard to keep it?

> . but just until december.

As mentioned in https://github.com/dipy/dipy/pull/2715#discussion_r1348580722,
- dropping a Python version should have been announced in the previous release.
- I am fine if from now one we will stick to the Scientific Python specs, but that should be made clear in the docs, and for now, keep 3.8 until the next release or EOL.

Reply:
> Python3.8 end of life is in less than 1y. It is not a good idea to stop it at the same time as python EOL. if we do it, many user will still use a version without security patch.

Still unclear; the Python folks will patch versions as they maintain them. It is up to the users to use the latest patched version; that is not DIPY's task.

>  Better to stop it before and let them the time to migrate before the EOL.

But users have to be warned beforehand.

Reply:
I recommend you to look at 

- https://numpy.org/neps/nep-0029-deprecation_policy.html
- https://github.com/nipy/nibabel/issues/803

this is what we approximately follow since a long time. Dropping python 3.8 in December does not sound so terrible. Numpy plan to do it in April, other package has already done it. Not sure what is the concern?  Yes documentation need to be updated concerning this topic and ok for announcing it.
Like I said above, we will keep it for this release and announce that we plan to drop it in the december release. 

@arokem, any opinion concerning this point ?

Reply:
My opinion is that dropping 3.8 now is not a big deal, but that our documentation should also state explicitly that we are following https://scientific-python.org/specs/spec-0000/, so that users are not caught unawares. In fact, according to that, we should also be dropping 3.9 support pretty soon. So, dropping 3.8 now seems really OK.

Reply:
Relevant to the discussion: https://endoflife.date/ and an example of a Python version dropping notice https://endoflife.date/numpy#python-support.